tests/lib/rules/no-unused-vars.js

```diff
@@ -333,6 +345,21 @@ ruleTester.run("no-unused-vars", rule, {
             ]
         },
 
+        // Destructured var without object rest sibling.
+        {
+            code: "const data = { type: 'coords', x: 1, y: 2 };\nconst { type } = data;",
+            options: [{ ignoreRestSiblings: true }],
+            parserOptions: {
+                ecmaVersion: 8,
+                ecmaFeatures: {
+                    experimentalObjectRestSpread: true
+                }
+            },
+            errors: [
+                { line: 2, column: 9, message: "'type' is assigned a value but never used." }
+            ]
+        },
```

Comment:
Can you add a couple tests where an unused variable is in the same declarator as a rest property, but it's not a sibling?

```js
var { foo: [unused], ...bar } = baz;
doSomethingWith(bar);
```

```js
var [unused, { ...bar }] = baz;
doSomethingWith(bar);
```

Both of these should result in an error getting reported.